### PR TITLE
Replace AESHash with MemHash

### PR DESCRIPTION
**Pull request by ashish-goswami**
Currently we are using AESHash(aeshash in runtime) which uses
aesenc instruction. This can crash restritto if it is running
on a hardware not having this instruction. Go runtime, at init
decides to use memhash if this instruction is not available.
memhash is independent of hardware. memhash on every call,
checks if aeshash can be calculated(based on the availabilty of
aesenc intruction), if yes it uses aeshash. If not, it uses normal flow(memhash).
This commit changes AESHash to MemHash.

Fixes: https://github.com/dgraph-io/badger/issues/980

---
This change is [Reviewable](https://reviewable.io/reviews/dgraph-io/ristretto/37)

### CODE REVIEW

File: z/rtutil.go [hunk 1]
```diff
@@ -40,21 +40,23 @@ type stringStruct struct {
 }
 
 //go:noescape
-//go:linkname aeshash runtime.aeshash
-func aeshash(p unsafe.Pointer, h, s uintptr) uintptr
+//go:linkname memhash runtime.memhash
+func memhash(p unsafe.Pointer, h, s uintptr) uintptr
 
-// AESHash is the hash function used by map, it utilizes available hardware instructions.
+// MemHash is the hash function used by map, it utilizes available hardware instructions(behaves as
```
Comment (jarifibrahim) [inline z/rtutil.go]: `go maps` instead of `map` would make more sense.

File: z/rtutil.go [hunk 2]
```diff
@@ -40,21 +40,23 @@ type stringStruct struct {
...
 // NOTE: The hash seed changes for every process. So, this cannot be used as a persistent hash.
-func AESHash(data []byte) uint64 {
+func MemHash(data []byte) uint64 {
 	ss := (*stringStruct)(unsafe.Pointer(&data))
-	return uint64(aeshash(ss.str, 0, uintptr(ss.len)))
+	return uint64(memhash(ss.str, 0, uintptr(ss.len)))
 }
 
-// AESHashString is the hash function used by map, it utilizes available hardware instructions.
+// MemHashString is the hash function used by map, it utilizes available hardware instructions
```
Comment (jarifibrahim) [inline z/rtutil.go]: `used by go maps`

Comment (ashish-goswami) [reply z/rtutil.go]: Done.

File: z/rtutil.go (hunk 1 above)
Comment (ashish-goswami) [reply z/rtutil.go]: Done.

**Comment by ashish-goswami** [comment]
GODEBUG=cpu.aes=off, disables aeshash.

**Pull request merged**